internal/identify/identify.go

```diff
@@ -62,64 +67,103 @@ type scrapeResult struct {
 	source ScraperSource
 }
 
-func (t *SceneIdentifier) scrapeScene(ctx context.Context, scene *models.Scene) (*scrapeResult, error) {
+func (t *SceneIdentifier) scrapeScene(ctx context.Context, txnManager txn.Manager, scene *models.Scene) (*scrapeResult, error) {
 	// iterate through the input sources
 	for _, source := range t.Sources {
 		// scrape using the source
-		scraped, err := source.Scraper.ScrapeScene(ctx, scene.ID)
+		results, err := source.Scraper.ScrapeScenes(ctx, scene.ID)
 		if err != nil {
 			logger.Errorf("error scraping from %v: %v", source.Scraper, err)
 			continue
 		}
 
-		// if results were found then return
-		if scraped != nil {
-			return &scrapeResult{
-				result: scraped,
-				source: source,
-			}, nil
+		if len(results) > 0 {
+			options := t.getOptions(&source)
+			if len(results) > 1 && *options.SkipMultipleMatches {
+				if options.SkipMultipleMatchTag != nil && len(*options.SkipMultipleMatchTag) > 0 {
+					// Tag it with the multiple results tag and ignore
+					err := t.addTagToScene(ctx, txnManager, scene, options.SkipMultipleMatchTag)
+					if err != nil {
+						return nil, err
+					}
+					return nil, nil
+				}
+			} else {
+				// if results were found then return
+				return &scrapeResult{
+					result: results[0],
+					source: source,
+				}, nil
+			}
 		}
 	}
 
 	return nil, nil
 }
 
+// Returns a MetadataOptions object with any default options overwritten by source specific options
+func (t *SceneIdentifier) getOptions(source *ScraperSource) MetadataOptions {
+	options := t.DefaultOptions
```

Comment:
This should dereference `t.DefaultOptions` so that the original isn't modified. It's passed as a pointer, so all the stuff below is modifying it.